src/app/routes/applications/new-role/components/role-field/role-field.component.ts

```diff
@@ -0,0 +1,183 @@
+import { Component, Input, EventEmitter, ChangeDetectorRef, Output } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { MatTableDataSource } from '@angular/material/table';
+
+const FIELD_TYPES = [
+  'text', 'number', 'date', 'boolean'
+];
+
+@Component({
+  selector: 'app-role-field',
+  templateUrl: './role-field.component.html',
+  styleUrls: ['./role-field.component.scss']
+})
+export class RoleFieldComponent {
+
+  @Input() fieldsForm: FormGroup;
+  @Input() dataSource: MatTableDataSource<any>;
+  @Input() isChecking: boolean;
+
+  @Output() resetFormEvent = new EventEmitter<boolean>();
+  @Output() updateDataSourceEvent: EventEmitter<MatTableDataSource<any>> = new EventEmitter();
+  @Output() backEvent: EventEmitter<any> = new EventEmitter();
+  @Output() proceedConfirmDetailsEvent: EventEmitter<any> = new EventEmitter();
```

Comment:
Can we name it `confirm` / `next` / `proceed` or something like this?

Reply:
updated